src/main/webapp/app/overview/course-dashboard/course-exercise-lateness/course-exercise-lateness.component.html

```diff
@@ -0,0 +1,62 @@
+<div>
+    <div class="d-flex flex-wrap justify-content-between align-items-baseline gap-2">
+        <div class="d-flex gap-2 align-items-baseline">
+            <h4 jhiTranslate="artemisApp.courseStudentDashboard.exerciseLateness.title"></h4>
+            <jhi-help-icon text="artemisApp.courseStudentDashboard.exerciseLateness.info" />
+        </div>
+        @if (isDataAvailable) {
+            <!-- Legend-->
+            <div class="d-flex gap-3">
+                <div class="d-flex align-items-center gap-1">
+                    <span class="tooltip-label-color" [style.background-color]="YOUR_GRAPH_COLOR"></span>
+                    <span jhiTranslate="artemisApp.courseStudentDashboard.exerciseLateness.yourLatenessLabel"></span>
+                </div>
+                <div class="d-flex align-items-center gap-1">
+                    <span class="tooltip-label-color" [style.background-color]="AVERAGE_GRAPH_COLOR"></span>
+                    <span jhiTranslate="artemisApp.courseStudentDashboard.exerciseLateness.averageLatenessLabel"></span>
+                </div>
+            </div>
+        }
+    </div>
+    @if (isDataAvailable) {
+        <div #containerRef class="mt-2 justify-content-center">
+            <ngx-charts-line-chart
+                [view]="[containerRef.offsetWidth, 200]"
+                [results]="ngxData"
+                [scheme]="ngxColor"
+                [legend]="false"
+                [maxXAxisTickLength]="8"
+                [xAxis]="true"
+                [xAxisLabel]="'artemisApp.courseStudentDashboard.exerciseLateness.xAxisLabel' | artemisTranslate"
+                [showXAxisLabel]="true"
+                [yAxis]="true"
+                [yAxisLabel]="'artemisApp.courseStudentDashboard.exerciseLateness.yAxisLabel' | artemisTranslate"
+                [showYAxisLabel]="true"
+                [yScaleMin]="0"
+                [yScaleMax]="yScaleMax"
+            >
+                <ng-template #tooltipTemplate let-model="model">
+                    <div class="d-flex flex-column align-items-start py-2">
+                        <b class="mb-1">{{ model.extra.title }}</b>
+                        {{ model.series }}: {{ round(model.value, 1) }}%
+                    </div>
+                </ng-template>
+                <ng-template #seriesTooltipTemplate let-model="model">
+                    @if (!model.isEmpty) {
+                        <div class="d-flex flex-column align-items-start py-2">
+                            <b class="mb-1">{{ model[0].extra.title }}</b>
+                            @for (entry of model; track entry) {
+                                <div class="d-flex align-items-center gap-1">
+                                    <span class="tooltip-label-color" [style.background-color]="entry.color"></span>
+                                    {{ entry.series }}: {{ round(entry.value, 1) }}%
+                                </div>
+                            }
```

Comment:
It might be hard due to the nested structure, but would it possible to avoid these calls to `round` here?